src/Foundation/Commerce/code/Services/Account/AccountService.cs

```diff
@@ -16,60 +16,387 @@ namespace Wooli.Foundation.Commerce.Services.Account
         }
 
-        public Result<IEnumerable<Address>> UpdateAddress(string userName, Address address)
+        public Result<VoidResult> UpdateAddress(string userName, Address address)
         {
-            throw new NotImplementedException();
+            Assert.ArgumentNotNullOrEmpty(userName, nameof(userName));
+            Assert.ArgumentNotNull(address, nameof(address));
+
+            var result = new Result<VoidResult>();
+
+            var customerResult = this.GetCustomerByName(userName);
+
+            if (customerResult.Success)
+            {
+                var getPartiesResult = this.accountManager.GetParties(customerResult.Data);
+
+                if (getPartiesResult.Success)
+                {
+                    var party = getPartiesResult.Parties.FirstOrDefault(p => p.ExternalId == address.ExternalId);
+
+                    if (party != null)
+                    {
+                        address.ExternalId = party.ExternalId;
+                        address.PartyId = party.PartyId;
+
+                        var commerceParty = this.mapper.Map<Address, CommerceParty>(address);
+
+                        commerceParty.RegionName = this.GetSubdivisionModelName(address);
+
+                        var updatePartyResponse = this.accountManager.UpdateParties(
+                            customerResult.Data,
+                            new List<Party>
+                            {
+                                commerceParty
+                            });
+
+                        if (!updatePartyResponse.Success)
+                        {
+                            result.SetErrors(updatePartyResponse.SystemMessages.Select(sm => sm.Message).ToList());
+                        }
+                    }
+                    else
+                    {
+                        result.SetError("party not found");
+                    }
+                }
+                else
+                {
+                    result.SetErrors(getPartiesResult.SystemMessages.Select(sm => sm.Message).ToList());
+                }
+            }
+            else
+            {
+                result.SetErrors(customerResult.Errors);
+            }
+
+            return result;
         }
 
         public Result<VoidResult> ValidateEmail(string email)
         {
-            throw new NotImplementedException();
+            Assert.ArgumentNotNullOrEmpty(email, nameof(email));
+
+            var result = new Result<VoidResult>();
+
+            // commerce connect don't have separate method for user's email validation
+            var emailRegex = new Regex(@"\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*");
+            if (emailRegex.Match(email).Success)
+            {
+                var getUserResult = this.accountManager.GetUsers(email: email);
+
+                if (getUserResult.CommerceUsers.FirstOrDefault() != null)
+                {
+                    result.SetError("Email is already in use!");
```

Comment:
Remove `!`

Reply:
Left logic like before, this massage will be on front end